packages/sentry_integrations_client/sentry_integrations_client/client.py

```diff
@@ -0,0 +1,145 @@
+"""gRPC-Web client for Sentry integrations service using Sonora."""
+
+import hashlib
+import hmac
+import ssl
+
+import urllib3
+from sonora.client import UnaryUnaryCall, UnaryUnaryMulticallable, WebChannel
+
+
+class AuthUnaryUnaryMulticallable(UnaryUnaryMulticallable):
+    """
+    Custom UnaryUnaryMulticallable that injects auth headers.
+
+    This extends Sonora's UnaryUnaryMulticallable to automatically inject
+    authentication headers (token or HMAC) into every request.
+    """
+
+    def __init__(
+        self,
+        session,
+        url,
+        path,
+        request_serializer,
+        response_deserializer,
+        auth_token: str | None = None,
+        hmac_secret: str | None = None,
+    ):
+        super().__init__(session, url, path, request_serializer, response_deserializer)
+        self._auth_token = auth_token
+        self._hmac_secret = hmac_secret
+        self._path = path
+
+    def with_call(self, request, timeout=None, metadata=None):
+        """Override to inject auth headers into metadata."""
+        # Start with base metadata
+        call_metadata = self._metadata.copy()
+
+        # Inject auth headers
+        if self._auth_token:
+            # Token-based authentication
+            call_metadata.append(("X-Grpc-Auth-Token", self._auth_token))
```

Comment:
We have something similar for the taskworker clients, it would be ideal to have one implementation of grpc request signing.